packages/grafana-data/src/text/sanitize.ts

```diff
@@ -54,15 +54,36 @@ const sanitizeTextPanelWhitelist = new xss.FilterXSS({
  * Note that sanitized tags will be removed, such as "<script>".
  * We don't allow form or input elements.
  */
-export function sanitize(unsanitizedString: string): string {
+export function sanitize(unsanitizedString: string, allowTargetBlank = false): string {
   try {
-    return DOMPurify.sanitize(unsanitizedString, {
+    let cfg: DOMPurify.Config = {
       USE_PROFILES: { html: true },
       FORBID_TAGS: ['form', 'input'],
-    });
+    };
+
+    if (allowTargetBlank) {
+      // hook to handle anchor elements with target="_blank"
+      DOMPurify.addHook('afterSanitizeAttributes', function (node) {
+        if (node.tagName === 'A' && node.getAttribute('target') === '_blank') {
```

Comment:
Could we simplify this by always setting `noopener` and `noreferrer` in the `rel` attribute? Something like:

```
        if (node.tagName === 'A' && node.getAttribute('target') === '_blank') {
          node.setAttribute('rel', 'noopener noreferrer');
        }
```

Reply:
@KristianGrafana do you think we should add it generally, not just for annotations? I added the flag here https://github.com/grafana/grafana/pull/106301/commits/353ad2eb032a067d851d9411eca278bb5f0a7ff0, as the original issue was reported for annotations. I wasn't sure what's the best approach/ how much we want to expose.

Reply:
@KristianGrafana I removed the flag and the hook - because we don't have access to the node anymore, I had to create a placeholder element. Let me know what you think!